### DISCUSSION TITLE
Homespan with LED and Real Button

### DISCUSSION THREAD

**Discussion by victor00000** [HomeSpan/HomeSpan#314] [Q&A]
I just using the example to learn how to use this library.

I am using example 05-working LED and using spanbutton() function.  My problem is, I need to push the button twice to trigger the LED. When I push the button continuously, it run OK. ON OFF ON OFF, it is good.  But if I leave it for 3 seconds,  I need to push the button twice to trigger.
Homekit switch is fine.

Did I missing anything?

here is my code
////////////////////////////////////
struct DEV_LED : Service::LightBulb {     
  int ledPin;        
  int powerPin;     
  SpanCharacteristic *power;      
  DEV_LED(int ledPin, int powerPin) : Service::LightBulb(){
    new SpanButton(powerPin);     
    power=new Characteristic::On(); 
    this->ledPin=ledPin;    
    this->powerPin=powerPin;  
    pinMode(ledPin,OUTPUT);    
    
  } // end constructor

  boolean update(){            
    digitalWrite(ledPin,power->getNewVal());     
    Serial.println("homekit");
    return(true);
  } // update

    void button(int pin, int pressType) override {
          if(pin==powerPin){
             if(pressType==SpanButton::SINGLE){ 
               power->setVal(1-power->getVal());         
               digitalWrite(ledPin,power->getNewVal());  
               Serial.println("button");
             }
          }
    }
};

**Comment by HomeSpan** [comment]
Are you using a normally-open switch (i.e. one that you need to press to close the circuit) that connects the pin directly to ground?

**Reply by victor00000** [reply]
it is a tactical switch, something like this.
[https://hken.rs-online.com/web/p/tactile-switches/6821131?cm_mmc=HK-PLA-DS3A-_-google-_-PLA_HK_EN_Switches_Whoop-_-(HK:Whoop!)+Tactile+Switches-_-6821131&matchtype=&aud-827186183686:pla-390435915031&gclid=Cj0KCQjwg_iTBhDrARIsAD3Ib5gGoVIRJ3OUZXnePEbYwNKTYBamDtjVMlK4fjSF95z36AdY9KQyISIaAotmEALw_wcB&gclsrc=aw.ds](url)
it is grounded and work well with Arduino IDE built-in example

**Reply by Xorlent** [reply]
Are you seeing the code within button() run?  If not, try putting a .1uF capacitor in parallel across the switch and see if the behavior changes.
